libs/ledger/src/chain/block.cpp

```diff
@@ -87,7 +87,7 @@ void Block::UpdateDigest()
   // Generate hash stream
   serializers::MsgPackSerializer buf;
   buf << previous_hash << merkle_hash << block_number << miner << log2_num_lanes << timestamp
-      << tx_merkle_tree.root();
+      << tx_merkle_tree.root() << block_entropy;
```

Comment:
Needed so that block entropy is part of hash + signature (avoid man in the middle attack on your blocks)